src/screens/settings-screen.tsx

```diff
@@ -1,63 +1,208 @@
-const Separator = () => <View style={styles.separatorStyle} />;
+	const dispatch = useTypedDispatch();
+
+	const alertOnReset = (target: string) => {
+		Alert.alert(
+			`Resetting ${target}`,
+			`Do you really want to reset ${target}?`,
+			[
+				{
+					text: "Cancel",
+				},
+				{
+					text: "Reset",
+					onPress: async () => {
+						if (target === "UserID") {
+							await dispatch(resetUser());
+							await dispatch(identifyUser());
+						}
+						if (target === "maptile cache") {
+							deleteTileCacheDirectory();
+							makeTileCacheDirectory();
+						}
+					},
+				},
+			]
+		);
+	};
+
+	let index = 0;
+	const trackFreq = [
+		{ key: index++, component: <Text>1 second</Text>, label: 1000 },
+		{ key: index++, component: <Text>5 seconds</Text>, label: 5000 },
+		{ key: index++, component: <Text>10 seconds</Text>, label: 10000 },
+		{ key: index++, component: <Text>30 seconds</Text>, label: 30000 },
+		{ key: index++, component: <Text>1 minute</Text>, label: 60000 },
+	];
+
+	index = 0;
+	const sendFreq = [
+		{ key: index++, component: <Text>10 seconds</Text>, label: 10000 },
+		{ key: index++, component: <Text>30 seconds</Text>, label: 30000 },
+		{ key: index++, component: <Text>1 minute</Text>, label: 60000 },
+		{ key: index++, component: <Text>5 minutes</Text>, label: 300000 },
+		{ key: index++, component: <Text>10 minutes</Text>, label: 600000 },
+	];
+
+	index = 0;
+	const tileLifetime = [
+		{ key: index++, component: <Text>12 hours</Text>, label: 43200000 },
+		{ key: index++, component: <Text>24 hours</Text>, label: 86400000 },
+		{ key: index++, component: <Text>48 hours</Text>, label: 172800000 },
+	];
+
+	const settingsData: SettingsData = [
+		{
+			type: "SECTION",
+			header: "Navigation".toUpperCase(),
+			footer:
+				"Change tracking and sending frequencies to save battery life and turn the app to offline mode",
+			rows: [
+				{
+					title: "Waypoint tracking frequency",
+					renderAccessory: () => (
+						<ModalSelector
+							data={trackFreq}
+							initValue={currTrack.toString()}
+							onChange={async (option) => {
+								await dispatch(setInterval(option.label, true));
+							}}
+						/>
+					),
+				},
+				{
+					title: "Waypoint sending frequency",
+					renderAccessory: () => (
+						<ModalSelector
+							data={sendFreq}
+							initValue={currSend.toString()}
+							onChange={async (option) => {
+								await dispatch(setInterval(option.label, false));
+							}}
+						/>
+					),
+				},
+				{
+					title: "Offline mode",
+					renderAccessory: () => <SettingsToggle />,
+				},
+			],
+		},
+		{
+			type: "SECTION",
+			header: "Map cache".toUpperCase(),
+			footer:
+				"Delete cached map tiles to clear space and change their lifetime",
+			rows: [
+				{
+					title: "Maptile lifetime",
+					renderAccessory: () => (
+						<ModalSelector
+							data={tileLifetime}
+							initValue="Lifetime"
+							onChange={(option) => {
+								alert(option.component);
```

Comment:
This currently renders `[object Object]`. I know this is still work in progress, but just wanted to point out just in case.